colossalai/inference/executor/rpc_worker.py

```diff
@@ -179,19 +181,18 @@ def _init_model(self, model_or_path: Union[nn.Module, str], model_policy: Policy
         """
 
         if isinstance(model_or_path, str):
-            # is_local = os.path.isdir(model_or_path)
+            is_local = os.path.isdir(model_or_path)
             try:
-                hf_config = AutoConfig.from_pretrained(model_or_path, trust_remote_code=True)
+                hf_config = AutoConfig.from_pretrained(model_or_path, trust_remote_code=True, torch_dtype=self.dtype)
                 arch = getattr(hf_config, "architectures")[0]
-                # NOTE(lry89757) Currently we load the model using transformers-api,
-                # but we will use lazy tensor and checkpoint io to accelerate
-                # the model load process in the future.
-                model = _SUPPORTED_MODELS[arch].from_pretrained(model_or_path, trust_remote_code=True)
-                # if is_local:
-                #     model = _SUPPORTED_MODELS[arch](hf_config)
-                # else:
-                #     # load the real checkpoint
-                #     model = _SUPPORTED_MODELS[arch].from_pretrained(model_or_path, trust_remote_code=True)
+                if is_local:
```

Comment:
For this method (`_init_model`), please also keep up to date with that (`init_model`) in `engine.py`. For example, the handling of condition of no policy is provided.

Reply:
Mentioned Above